appmap_remote.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+set -x
+
+LOG=petclinic.log
+java -ea -Dappmap.debug -javaagent:appmap.jar -jar target/spring-petclinic*.jar &> $LOG &
+JVM_PID=$!
+
+export WS_URL="http://localhost:8080"
+export OUTPUT_DIR="$1"
+export APPMAP_FILE_SUFFIX=".appmap.json"
+export IGNORE_RESPONSE_BODY="-o /dev/null"
+
+mkdir -p $OUTPUT_DIR
+
+printf 'getting set up'
+while [ -z "$(curl -sI "${WS_URL}" | grep 'HTTP/1.1 200')" ]; do
+    if ! kill -0 "${JVM_PID}" 2> /dev/null; then
+        printf '. failed!\n\nprocess exited unexpectedly:\n'
+        cat $LOG
+        exit 1
+    fi
+    printf '.'
+    sleep 2
+done
+printf ' done!\n'
+
+_curl() {
+    curl -H 'Accept: application/json' "$@"
+}
+
+start_recording() {
+    unset output
+    _curl -sXPOST "${WS_URL}/_appmap/record"
+}
+
+stop_recording() {
+    output=$(_curl -sXDELETE ${WS_URL}/_appmap/record)
+    [ ! "$#" -eq "1" ] && exit 1
```

Comment:
Is this actually trying to check the number of arguments, or the exit status of `curl`? If you're trying to check the exit status of curl, you want to use `$?` rather than `$#`.

Reply:
it's checking number of args. I haven't added checks  for whether appmap agent is loaded or not and other sanity checks as we will be the maintaining this file.

Reply:
Ok. In that case, I think it would be better to write an error message, too:
```
if [[ $# != 1 ]]; then
  echo "Wrong number of arguments, $# ($@)"
  exit 1
fi
```